docs/modules/ROOT/pages/erc20.adoc

```diff
@@ -11,410 +11,616 @@ The `ERC20.cairo` contract implements an approximation of https://eips.ethereum.
+  *** <<internal_transfer,_transfer>>
+  *** <<internal_spend_allowance,_spend_allowance>>
 
 == Interface
 
-[,cairo]
-----
-@contract_interface
-namespace IERC20 {
-    func name() -> (name: felt) {
-    }
-
-    func symbol() -> (symbol: felt) {
-    }
-
-    func decimals() -> (decimals: felt) {
-    }
-
-    func totalSupply() -> (totalSupply: Uint256) {
-    }
-
-    func balanceOf(account: felt) -> (balance: Uint256) {
-    }
-
-    func allowance(owner: felt, spender: felt) -> (remaining: Uint256) {
-    }
-
-    func transfer(recipient: felt, amount: Uint256) -> (success: felt) {
-    }
-
-    func transferFrom(sender: felt, recipient: felt, amount: Uint256) -> (success: felt) {
-    }
-
-    func approve(spender: felt, amount: Uint256) -> (success: felt) {
-    }
+[,rust]
+----
+#[starknet::interface]
+trait IERC20<TState> {
+    fn name(self: @TState) -> felt252;
+    fn symbol(self: @TState) -> felt252;
+    fn decimals(self: @TState) -> u8;
+    fn total_supply(self: @TState) -> u256;
+    fn balance_of(self: @TState, account: ContractAddress) -> u256;
+    fn allowance(self: @TState, owner: ContractAddress, spender: ContractAddress) -> u256;
+    fn transfer(ref self: TState, recipient: ContractAddress, amount: u256) -> bool;
+    fn transfer_from(
+        ref self: TState, sender: ContractAddress, recipient: ContractAddress, amount: u256
+    ) -> bool;
+    fn approve(ref self: TState, spender: ContractAddress, amount: u256) -> bool;
 }
 ----
 
 === ERC20 compatibility
 
-Although StarkNet is not EVM compatible, this implementation aims to be as close as possible to the ERC20 standard, in the following ways:
+Although Starknet is not EVM compatible, this implementation aims to be as close as possible to the ERC20 standard, in the following ways:
 
-* It uses Cairo's `uint256` instead of `felt`.
-* It returns `TRUE` as success.
-* It accepts a `felt` argument for `decimals` in the constructor calldata with a max value of 2{caret}8 (imitating `uint8` type).
+* It uses Cairo's `u256` instead of `felt252`.
 * It makes use of Cairo's short strings to simulate `name` and `symbol`.
 
 But some differences can still be found, such as:
 
-* `transfer`, `transferFrom` and `approve` will never return anything different from `TRUE` because they will revert on any error.
-* Function selectors are calculated differently between https://github.com/starkware-libs/cairo-lang/blob/7712b21fc3b1cb02321a58d0c0579f5370147a8b/src/starkware/starknet/public/abi.py#L25[Cairo] and https://solidity-by-example.org/function-selector/[Solidity].
+* `transfer`, `transfer_from` and `approve` will never return anything different from `true` because they will revert on any error.
+* Function selectors are calculated differently between https://github.com/starkware-libs/cairo/blob/7dd34f6c57b7baf5cd5a30c15e00af39cb26f7e1/crates/cairo-lang-starknet/src/contract.rs#L39-L48[Cairo] and https://solidity-by-example.org/function-selector/[Solidity].
```

Comment:
I'm not sure we should include this here. The fact that function selectors are computed differently is not directly related to the implementation of the ERC20 standard, because it says nothing about selectors, right? And also, using a different casing would give different selectors anyway. Casing may be included too.

Reply:
That's fair. The selector calculation is more of a network difference, not really specific to ERC20. And yeah, good call on including the casing difference